omnigibson/prims/rigid_prim.py

```diff
@@ -317,48 +340,87 @@ def set_position_orientation(self, position=None, orientation=None):
                 np.linalg.norm(orientation), 1, atol=1e-3
             ), f"{self.prim_path} desired orientation {orientation} is not a unit quaternion."
             orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-        self._rigid_prim_view.set_world_poses(positions=position, orientations=orientation)
+
+        if frame == "world" or frame == "scene":
+            self._rigid_prim_view.set_world_poses(positions=position, orientations=orientation)
+        else:
+            self._rigid_prim_view.set_local_poses(positions=position, orientations=orientation)
+
         PoseAPI.invalidate()
 
-    def get_position_orientation(self):
-        # Return cached pose if we're kinematic-only
-        if self.kinematic_only and self._kinematic_world_pose_cache is not None:
-            return self._kinematic_world_pose_cache
+    def get_position_orientation(self, frame: Literal["world", "scene", "parent"] = "world"):
+        """
+        Gets prim's pose with respect to the specified frame.
 
-        pos, ori = self._rigid_prim_view.get_world_poses()
+        Args:
+            frame (Literal): frame to get the pose with respect to. Default to world. parent frame
+            get position relative to the object parent. scene frame get position relative to the scene.
 
-        assert np.isclose(
-            np.linalg.norm(ori), 1, atol=1e-3
-        ), f"{self.prim_path} orientation {ori} is not a unit quaternion."
+        Returns:
+            2-tuple:
+                - 3-array: (x,y,z) position in the specified frame
+                - 4-array: (x,y,z,w) quaternion orientation in the specified frame
+        """
 
-        pos = pos[0]
-        ori = ori[0][[1, 2, 3, 0]]
-        if self.kinematic_only:
-            self._kinematic_world_pose_cache = (pos, ori)
-        return pos, ori
+        if frame == "world" or frame == "scene":
 
-    def set_local_pose(self, position=None, orientation=None):
-        # Invalidate kinematic-only object pose caches when new pose is set
-        if self.kinematic_only:
-            self.clear_kinematic_only_cache()
-        if position is not None:
-            position = np.asarray(position)[None, :]
-        if orientation is not None:
-            orientation = np.asarray(orientation)[None, [3, 0, 1, 2]]
-        self._rigid_prim_view.set_local_poses(position, orientation)
-        PoseAPI.invalidate()
+            if self.kinematic_only and self._kinematic_world_pose_cache is not None:
+                return self._kinematic_world_pose_cache
```

Comment:
This is wrong, right? The cache is world pose. If frame == "scene", we need to transform it into scene frame.
Maybe we also need a scene pose cache?